collect_stata/write_json.py

```diff
@@ -60,79 +74,82 @@ def uni_cat(elem, file_csv):
 
 
 def uni_string():
-    """Generate dict with frequencies for nominal variables
+    """Generate dict with frequencies for nominal variables.
 
-    Output:
-    OrderedDict
+    Returns:
+        OrderedDict(...) (dict): Empty placeholder for frequencies,
+        labels, labels_de, missings and values for nominal variables.
     """
 
     return OrderedDict(frequencies=[], labels=[], labels_de=[], missings=[], values=[])
 
 
 def uni_number():
-    """Generate dict with frequencies for numerical variables
+    """Generate dict with frequencies for numerical variables.
 
-    Output:
-    OrderedDict
+    Returns:
+        OrderedDict(...) (dict): Empty placeholder for frequencies,
+        labels, labels_de, missings and values for numerical variables.
     """
 
     return OrderedDict(frequencies=[], labels=[], labels_de=[], missings=[], values=[])
 
 
-def stats_cat(elem, file_csv):
-    """Generate dict with statistics for categorical variables
+def stats_cat(elem, data):
+    """Generate dict with statistics for categorical variables.
 
-    Input:
-    elem: dict
-    file_csv: pandas DataFrame
+    Args:
+        elem (dict): Name, label, type and values of categorical variables.
+        data (pandas.DataFrame): Datatable of imported data.
 
-    Output:
-    dict
+    Returns:
+        {...} (dict): Number of valid and invalid values.
     """
 
-    total = file_csv[elem["name"]].size
-    invalid = int(file_csv[elem["name"]].isnull().sum()) + int(
-        sum(n < 0 for n in file_csv[elem["name"]])
+    total = data[elem["name"]].size
+    invalid = int(data[elem["name"]].isnull().sum()) + int(
+        sum(n < 0 for n in data[elem["name"]])
     )
     valid = total - invalid
 
     return {"valid": valid, "invalid": invalid}
 
 
-def stats_string(elem, file_csv):
-    """Generate dict with statistics for nominal variables
+def stats_string(elem, data):
+    """Generate dict with statistics for nominal variables.
 
-    Input:
-    elem: dict
-    file_csv: pandas DataFrame
+    Args:
+        elem (dict): Name, label and type of nominal variables.
+        data (pandas.DataFrame): Datatable of imported data.
 
-    Output:
-    dict
+    Returns:
+        {...} (dict): Number of valid and invalid values.
```

Comment:
You would not specify names for return values. The name of a variable that is returned is irrelevant for the interaction with a function's interface.
Only type and what the returned value represents, should be documented.
Type should be specified with type hints.
Here this would be `-> dict`.
This would mean that the function should look soething like this:
```python
def stats_string(elem:dict , data: dict) -> dict:
    """Generate dict with statistics for nominal variables.

    Args:
        elem: Name, label and type of nominal variables.
        data: Datatable of imported data.

    Returns:
        Number of valid and invalid values.
```
However, neither the name of the function nor the docstring itself tell me anything about what the function actually does. What are "statistics for nominal variables".
It seems to me, that the function counts valid and invalid occurrences of something.